ts/navigation/components/HeaderFirstLevelHandler.tsx

```diff
@@ -163,49 +212,69 @@ export const HeaderFirstLevelHandler = ({ currentRouteName }: Props) => {
         if (isNewWalletSectionEnabled) {
           return {
             title: I18n.t("wallet.wallet"),
-            type: "singleAction",
             firstAction: helpAction,
-            testID: "wallet-home-header-title"
+            testID: "wallet-home-header-title",
+            ...(isSettingsVisibleAndHideProfile
+              ? {
+                  type: "twoActions",
+                  secondAction: settingsAction
+                }
+              : { type: "singleAction" })
           };
         }
         return {
           title: I18n.t("wallet.wallet"),
-          type: "twoActions",
           firstAction: helpAction,
           backgroundColor: "dark",
           testID: "wallet-home-header-title",
-          secondAction: {
-            icon: "add",
-            accessibilityLabel: I18n.t("wallet.accessibility.addElement"),
-            onPress: presentWalletHomeHeaderBottomsheet,
-            testID: "walletAddNewPaymentMethodTestId"
-          }
+          ...(isSettingsVisibleAndHideProfile
+            ? {
+                type: "threeActions",
+                secondAction: settingsAction,
+                thirdAction: walletAction
+              }
+            : {
+                type: "twoActions",
+                secondAction: walletAction
+              })
         };
       case ROUTES.PAYMENTS_HOME:
         return {
           title: I18n.t("features.payments.title"),
-          type: "singleAction",
-          firstAction: helpAction
+          firstAction: helpAction,
+          ...(isSettingsVisibleAndHideProfile
+            ? {
+                type: "twoActions",
+                secondAction: settingsAction
+              }
+            : { type: "singleAction" })
         };
       case MESSAGES_ROUTES.MESSAGES_HOME:
       default:
         return {
           title: I18n.t("messages.contentTitle"),
-          type: "twoActions",
           firstAction: helpAction,
-          secondAction: {
-            icon: "search",
-            accessibilityLabel: I18n.t("global.accessibility.search"),
-            onPress: messageSearchCallback
-          }
+          ...(isSettingsVisibleAndHideProfile
+            ? {
+                type: "threeActions",
+                secondAction: settingsAction,
```

Comment:
Be aware that, before leaving the message section, we should check for the archiving processing:
- if it is archiving/restoring, it is better not to navigate at all
- if it is scheduling the archiviation/restoring, it is better to cancel the scheduling, in order for the bottom tab bar to re-appear.

You can look at the `messageSearchCallback` for such cases

Reply:
Yes, I will make the change, thank you very much!

Reply:
Done!